### PR TITLE
[Refactoring] Make util classes final and add private constructors

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Purpose
- Add private constructor to utility classes
- Mark util classes final
https://sonarcloud.io/organizations/ballerina-platform/rules?open=java%3AS1118&rule_key=java%3AS1118
https://sonarcloud.io/organizations/ballerina-platform/rules?open=java%3AS2440&rule_key=java%3AS2440

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/ls-extensions/modules/trigger-service/src/main/java/io/ballerina/trigger/entity/Constants.java [hunk 1]
```diff
@@ -20,6 +20,9 @@
  *
  * @since 2.0.0
  */
-public class Constants {
+public final class Constants {
     public static final String CAPABILITY_NAME = "ballerinaTrigger";
```
Comment (HindujaB) [inline misc/ls-extensions/modules/trigger-service/src/main/java/io/ballerina/trigger/entity/Constants.java:24]:
```suggestion
public final class Constants {

    public static final String CAPABILITY_NAME = "ballerinaTrigger";
```

File: misc/maven-resolver/src/main/java/org/ballerinalang/maven/Utils.java [hunk 2]
```diff
@@ -22,7 +22,10 @@
 /**
  * Utilities for Maven dependency resolving.
  */
-public class Utils {
+public final class Utils {
+    private Utils() {
```
Comment (HindujaB) [inline misc/maven-resolver/src/main/java/org/ballerinalang/maven/Utils.java:26]:
```suggestion
public final class Utils {

    private Utils() {
```

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/utils/AssertionUtils.java [hunk 3]
```diff
@@ -30,13 +30,16 @@
  *
  * @since 2.0.0
  */
-public class AssertionUtils {
+public final class AssertionUtils {
     private static final Boolean isWindows = System.getProperty("os.name").toLowerCase(Locale.getDefault())
```
Comment (HindujaB) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/utils/AssertionUtils.java:34]:
```suggestion
public final class AssertionUtils {

    private static final Boolean isWindows = System.getProperty("os.name").toLowerCase(Locale.getDefault())
```

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/utils/ResponseReader.java [hunk 4]
```diff
@@ -31,9 +31,12 @@
 /**
  * Utility class for managing responses.
  */
-public class ResponseReader {
+public final class ResponseReader {
     private static final Logger LOG = LoggerFactory.getLogger(ResponseReader.class);
```
Comment (HindujaB) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/utils/ResponseReader.java:35]:
```suggestion
public final class ResponseReader {

    private static final Logger LOG = LoggerFactory.getLogger(ResponseReader.class);
```

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/utils/AssertionUtils.java (hunk 3 above)
Comment (HindujaB) [reply tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/utils/AssertionUtils.java:34]: Shall we check all other places and add new lines after the class header?

**Comment by HindujaB** [comment]
For readability, it is recommended to have a blank line after the class name header. 
https://raw.githubusercontent.com/wso2/product-is/master/docs/wso2_codestyle.xml
We can modify the classes in the PR accordingly to give all of them a similar format.

**Comment by Shadow-Devil** [comment]
@HindujaB I fixed the newline after class header for all classes that were previously in this PR.
Should I also fix it for any other class (all non util classes) or is this fine for now?

**Comment by HindujaB** [comment]
> @HindujaB I fixed the newline after the class header for all classes that were previously in this PR. Should I also fix it for any other class (all non-util classes) or is this fine for now? There are ~450 other classes where this could be applied.

It is better to get them changed as well. Since the PR is getting larger, I recommend creating a new PR for this and address there.

**Review by HindujaB — DISMISSED** [review]
LGTM

**Comment by Shadow-Devil** [comment]
> For readability, it is recommended to have a blank line after the class name header.
> https://raw.githubusercontent.com/wso2/product-is/master/docs/wso2_codestyle.xml
> We can modify the classes in the PR accordingly to give all of them a similar format.

@HindujaB 
In the codestyle that you're referring to, there is also a rule that every method should start with a blank line, which I don't see anywhere in the project. 
```
    <option name="BLANK_LINES_BEFORE_METHOD_BODY" value="1" />
```
I would argue that this is unnecessary whitespace and should not be applied. What do you think?

**Review by gimantha — APPROVED** [review]

**Pull request merged**